src/payment/strategies/sdk/strategy.ts

```diff
@@ -0,0 +1,76 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import { handler } from './handler';
+
+/*
+    BIG TODO: How do we handle the return journey of redirects?
+*/
+
+/*
+    This strategy would be used for all PPSDK methods, other than:
+        - ones requiring our credit card hosted field
+        - 'non-standard' SDK methods
+
+    Key points:
+        - The use of a "PPSDK Response Handler", which would
+            (1) discern different code paths based on success/failure/error/continue
+                responses from our checkout/processing endpoint (received via HostedFormFactory above)
+            (2) not return control to the strategy until all continue steps are exhausted
+            (3) attempt to handle any recoverable failure responses(?)
+            (4) translate error and un-recoverable failure responses into Checkout SDK errors
+                (e.g. those from src/common/error/errors/index.ts)
+                before throwing
+            (5) Delegate and await the resolving of relevant(?) continue steps to checkout-js
+            (6) Return a resolving promise from 'execute'
+                upon a success message from the checkout/processing endpoint
+
+*/
+
+export class Strategy implements PaymentStrategy {
+    constructor(
+        protected _store: CheckoutStore,
+        protected _orderActionCreator: OrderActionCreator
+    ) {}
+
+    initialize(): Promise<InternalCheckoutSelectors> {
+        // Nothing to be done here?
+        return Promise.resolve(this._store.getState());
+    }
+
+    deinitialize(_options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        // Nothing to be done here?
+        return Promise.resolve(this._store.getState());
+    }
+
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        // this strategy doesn't really know what the view will send it
+        // it will just pass it on to the processing/checkout endpoint
+
+        if (!options?.gatewayId || !options?.methodId) {
+            // throw? reject?
+            return Promise.reject();
+        }
+
+        const { gatewayId: paymentProvider, methodId } = options;
+
+        const jsonBody = JSON.stringify(payload);
+
+        const initialResponse = await fetch(`/payment/${paymentProvider}.${methodId}`, {
```

Comment:
In this the new payment endpoint? If so, does it require an order to be created first? If it is not the new payment endpoint, what is it? And at what point do we call the new payment endpoint to start/continue a payment transaction?

Reply:
This is a new endpoint, one that would allow us to get the next bit of orchestration from the BE. (https://github.com/bigcommerce/api-schema/pull/446#discussion_r603002341)

You're right, it would need an order to be created first.

I think I may be misusing it here, will refactor into a singular Strategy and take another look

Reply:
The non-hosted 'branch' has been effectively removed from this PR, hopefully making the comparison moot and this bit of conversation resolvable 😄